include/znc/Message.h

```diff
@@ -39,6 +39,25 @@ class CChan;
 class CClient;
 class CIRCNetwork;
 
+/**
+ * Here is a small explanation of how messages on IRC work, and how you can use
+ * this class to get useful information from the parsed message.
+ *
+ * @time=some-timestamp :server.network.net 366 something #channel :End of /NAMES list.
+ * tags                                     0   1         2        3
+ *
+ * `tags` is the IRCv3 tags associated with the message, obtained with GetTag(s).
+ * `0` is the command, which is obtained via GetCommand, or with GetParam(0)
+ * `1` is the first parameter index, which usually includes a sender or a recipient.
+ * `2` is the target or element the message is about.
+ * `3` is the message.
```

Comment:
This depends on the message type. What you describe here, is true only for 366 numeric, but it sounds like parameters 0-3 have this exact meaning always. Not every message has exactly this number of parameters.